### PR TITLE
:bento: try change application metrics to counter

### PR DESCRIPTION
**Pull request by songxiaosheng**
## What is the purpose of the change

:bento: try change application metrics to counter

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by AlbumenJ** [comment]
image

**Review by AlbumenJ — APPROVED** [review]
LGTM
@wxbty PTAL

File: dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/model/sample/CounterMetricSample.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    private final T value;
+
+    public CounterMetricSample(String name, String description, Map<String, String> tags,
+                               MetricsCategory category, T value ) {
+        super(name, description, tags, Type.COUNTER, category);
+        this.value = value;
+    }
+
+    public CounterMetricSample(String name, String description, Map<String, String> tags,   MetricsCategory category,
```
Comment (wxbty) [inline dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/model/sample/CounterMetricSample.java:33]: Is this useless?

Comment (songxiaosheng) [reply dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/model/sample/CounterMetricSample.java:33]: yes we cloud use it in the future

**Comment by songxiaosheng** [comment]
counter will show type in endpoints such as follow 
image

File: dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/report/AbstractMetricsReporter.java [hunk 2]
```diff
@@ -157,19 +159,17 @@ public void refreshData() {
...
 
                             Gauge.builder(gaugeSample.getName(), gaugeSample.getValue(), gaugeSample.getApply())
                                 .description(gaugeSample.getDescription()).tags(tags).register(compositeRegistry);
                             break;
                         case COUNTER:
+                            CounterMetricSample counterMetricSample = (CounterMetricSample) sample;
+                            FunctionCounter.builder(counterMetricSample.getName(),  counterMetricSample.getValue(),
+                                    Number::doubleValue).description(counterMetricSample.getDescription())
+                                .tags(getTags(counterMetricSample))
+                                .register( MetricsGlobalRegistry.getCompositeRegistry());
```
Comment (wxbty) [inline dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/report/AbstractMetricsReporter.java:172]: Use compositeRegistry to be consistent with the above

Comment (wxbty) [reply dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/report/AbstractMetricsReporter.java:172]: A few minor issues, no issues with functionality

Comment (songxiaosheng) [reply dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/report/AbstractMetricsReporter.java]: > Use compositeRegistry to be consistent with the above

done

**Review by wxbty — APPROVED** [review]

**Review by AlbumenJ — APPROVED** [review]

**Pull request merged**